dist/plugins/DefaultPlugin/resources/analysisTools/roddyTools/wrapInScript.sh

```diff
@@ -10,55 +20,63 @@
 # Cluster options (like i.e. PBS ) have to be parsed and set before job submission!
 # They will be ignored after the script is wrapped.
 
-# Perform some initial checks
-# Store the environment, store file locations in the env
-extendedLogs=`dirname $0`/extendedlogs
-extendedLogFile=${extendedLogs}/`basename ${0}`
-mkdir ${extendedLogs} &> /dev/null
-
-env > ${extendedLogFile}
-echo "" >> ${extendedLogFile}
-echo "Files in environment before source config" >> ${extendedLogFile}
-while IFS='=' read -r -d '' n v; do     [[ -r $v ]] && echo "$v -> "$(readlink -f "$v"); done < <(env -0)
-echo "" >> ${extendedLogFile}
-
-if [[ ${PARAMETER_FILE-false} != false ]]; then
-  while [[ ! -r ${PARAMETER_FILE} && ${waitCount-0} -lt 3 ]]; do sleep 5; waitCount=$((waitCount + 1)); done
-  [[ ! -r ${PARAMETER_FILE} && ${waitCount-0} -lt 3 ]] && echo "Roddy is setup to use job parameter files but the file ${PARAMETER_FILE} does not exist" && exit 199
-  source ${PARAMETER_FILE}
-fi
+dumpEnvironment() {
+    local message="${1:?No log message given}"
+    echo "$message"
+    while IFS='=' read -r -d '' n v; do     [[ -r $v ]] && echo "$v -> "$(readlink -f "$v"); done < <(env -0)
+    echo ""
+}
 
-[[ ${CONFIG_FILE-false} == false ]] && echo "The parameter CONFIG_FILE is not set but the parameter is mandatory!" && exit 200
+createToolVariableName() {
+    local varName="${1:?No variable name given}"
+    echo "$varName" | perl -ne 's/([A-Z])/_$1/g; print uc($_)'
+}
 
-waitCount=0
-while [[ ! -r ${CONFIG_FILE} && ${waitCount-0} -lt 3 ]]; do sleep 5; waitCount=$((waitCount + 1)); done
-[[ ! -f ${CONFIG_FILE} || ! -r ${CONFIG_FILE} ]] && echo "The configuration file ${CONFIG_FILE} does not exist or is not readable." && exit 200
+# Basic modules / environment support
+# Load the environment script (source), if it is defined. If the file is defined but the file not accessible exit with
+# code 200. Additionally, expose the used environment script path as ENVIRONMENT_SCRIPT variable to the wrapped script.
+runEnvironmentSetupScript() {
+    local envScriptVar="${TOOL_ID}EnvironmentScript"
+    if [[ -n "${!envScriptVar}" ]]; then
+        declare -gx ENVIRONMENT_SCRIPT="${!envScriptVar}"
```

Comment:
As far as I understand -g, this will make the variable available outside the function right?
Since when is supported in Bash?

Reply:
bash 4.2

Reply:
If we want the scripts to adhere to a specific version, we should set `shopt -o compat31` or similar (compat32, compat40, compat41; for some strange reason bash 4.2 does not offer a compat42 option 8-|; similary to set -eu should always be turned on, for minimal error checking/handling).

Reply:
... but we have not decided on any version.

Reply:
O.k. implemented `declare_xg ` to simulate this.